File: clients/desktop-client/src/main/java/ec/edu/grupo3/client/controller/AuthController.java
package ec.edu.grupo3.client.controller;

import ec.edu.grupo3.client.model.AuthModel;
import ec.edu.grupo3.client.view.LoginView;
import javafx.application.Platform;

public class AuthController {

    private final AuthModel authModel;
    private final LoginView loginView;
    private final Runnable onLoginSuccess;

    public AuthController(LoginView loginView, Runnable onLoginSuccess) {
        this.authModel = new AuthModel();
        this.loginView = loginView;
        this.onLoginSuccess = onLoginSuccess;

        setupEventHandlers();
    }

    private void setupEventHandlers() {
        loginView.getLoginButton().setOnAction(e -> handleLogin());
    }

    private void handleLogin() {
        String username = loginView.getUsername();
        String password = loginView.getPassword();

        if (username.isEmpty() || password.isEmpty()) {
            loginView.showError("Por favor complete todos los campos");
            return;
        }

        if (authModel.authenticate(username, password)) {
            onLoginSuccess.run();
        } else {
            loginView.showError("Usuario o contraseña incorrectos");
            loginView.clearFields();
        }
    }
}

File: clients/mobile-client/app/src/main/java/ec/edu/grupo3/mobile/controller/AuthController.kt
package ec.edu.grupo3.mobile.controller

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import ec.edu.grupo3.mobile.model.AuthRepository

class AuthController : ViewModel() {
    var username by mutableStateOf("")
        private set
    
    var password by mutableStateOf("")
        private set
    
    var errorMessage by mutableStateOf<String?>(null)
        private set
    
    var isAuthenticated by mutableStateOf(false)
        private set

    fun onUsernameChange(value: String) {
        username = value
        errorMessage = null
    }

    fun onPasswordChange(value: String) {
        password = value
        errorMessage = null
    }

    fun login() {
        if (AuthRepository.authenticate(username, password)) {
            errorMessage = null
            isAuthenticated = true
        } else {
            errorMessage = "Usuario o contraseña incorrectos"
        }
    }

    fun logout() {
        username = ""
        password = ""
        errorMessage = null
        isAuthenticated = false
    }
}